### PR TITLE
feat(component): added start and end icon to Badge

### PR DESCRIPTION
**Pull request by Wake1st**
# Added Start and End Icon to Badge Component

Icons used before and after the badge content, but still within the Badge itself. The padding and position had to be adjusted manually to allow the icons to fit neatly with center alignment along the horizontal axis. The Icon sizes used are small for the default sized badge and tiny for the small sized badge. Since the tiny sized icons are the same height as the small sized badge (16px), some of the icons barely fit.

Resolving #2125

Committer: Wake1st <joeltwakefield@gmail.com>

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/main/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [ ] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

**Comment by Wake1st** [comment]
Okie dokie, I did my best to follow the guidelines and recommendations, but if anything needs correction, please let me know.

**Comment by denkristoffer** [comment]
I think we should consider always using the `tiny` icon size in badge, what do you think @fabe?

**Comment by denkristoffer** [comment]
I took the liberty of collecting the examples in one 🙂

File: packages/components/badge/src/Badge/Badge.tsx [hunk 1]
```diff
@@ -2,52 +2,73 @@ import React from 'react';
...
         testId={testId}
-        display="inline-block"
-        className={cx(getBadgeStyles({ variant, size }), className)}
+        display="inline-flex"
+        className={cx(styles.badge({ variant, size }), className)}
         {...otherProps}
         ref={ref}
       >
+        {startIcon && iconContent(startIcon)}
         {children}
```
Comment (denkristoffer) [inline packages/components/badge/src/Badge/Badge.tsx:70]:
```suggestion
        <span>{children}</span>
```

**Review by denkristoffer — COMMENTED** [review]
Thanks for this @Wake1st! I've had a look and I think we can simplify the styling a bit, but this looks really great! 👏

File: packages/components/badge/src/Badge/Badge.styles.ts [hunk 2]
```diff
@@ -64,26 +64,52 @@ const sizeToStyles = ({ size }: { size: BadgeSize }): CSSObject => {
...
+  return css([
+    margin,
+    padding,
+    variant !== 'transparent' &&
+      hasChildren && {
+        '& svg': {
+          fill: 'currentColor',
+        },
+      },
+  ]);
```
[the remark this answers is not in the document]
Comment (denkristoffer) [reply packages/components/badge/src/Badge/Badge.styles.ts:95]: Let's see if I missed anything here 😅

**Comment by fabe** [comment]
> I think we should consider always using the `tiny` icon size in badge, what do you think @fabe?

Yes, this should only use the `tiny` size for the icon. Also, I'd opt for not rendering an icon if passed to the `small` variant of the Badge—we don't support an icon size that would make sense for it, at the moment.

**Comment by Wake1st** [comment]
I'll pass the color directly from the variant, unless its `'primary-filled'`, in which case the variant is `white`.
Also, I added some spacing, `spacing2Xs`, since the badge looks a little cramped without it.

Without passing the color:
image

Without spacing: 
image

**Comment by denkristoffer** [comment]
Really sorry but I blew up the git history here... Continued in #2265

**Pull request closed without merging**